frontend/src/scenes/data-warehouse/editor/multitabEditorLogic.tsx

```diff
@@ -336,24 +340,31 @@ export const multitabEditorLogic = kea<multitabEditorLogicType>([
 
             await dataWarehouseViewsLogic.asyncActions.createDataWarehouseSavedQuery({ name, query, types })
         },
-        reloadMetadata: async (_, breakpoint) => {
-            const model = props.editor?.getModel()
-            if (!model || !props.monaco) {
-                return
-            }
-            await breakpoint(300)
-            const query = values.queryInput
-            if (query === '') {
-                return
-            }
-
-            const response = await performQuery<HogQLMetadata>({
-                kind: NodeKind.HogQLMetadata,
-                language: HogLanguage.hogQL,
-                query: query,
+        saveAsInsight: async () => {
+            LemonDialog.openForm({
+                title: 'Save as new insight',
+                initialValues: {
+                    name: '',
+                },
+                content: (
+                    <LemonField name="name">
+                        <LemonInput data-attr="insight-name" placeholder="Please enter the new name" autoFocus />
+                    </LemonField>
+                ),
+                errors: {
+                    name: (name) => (!name ? 'You must enter a name' : undefined),
+                },
+                onSubmit: async ({ name }) => actions.saveAsInsightSubmit(name),
+            })
+        },
+        saveAsInsightSubmit: async ({ name }) => {
+            const insight = await insightsApi.create({
+                name,
+                query: props.sourceQuery,
+                saved: true,
             })
-            breakpoint()
-            actions.setMetadata(query, response)
+
+            router.actions.push(urls.insightView(insight.short_id))
```

Comment:
Adding the toast popup from https://github.com/PostHog/posthog/pull/26599 may be nice here